dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/AbstractOracleConnectorTest.java

```diff
@@ -44,4 +50,52 @@ public void buildProperties_success() {
     assertEquals(EXAMPLE_PASSWORD, actual.getProperty("password"));
     assertEquals(EXAMPLE_USER, actual.getProperty("user"));
   }
+
+  @Test
+  public void buildUrl_providedBoth_throwsException() {
+    when(arguments.getOracleServicename()).thenReturn("ORCLPDB");
+    when(arguments.getOracleSID()).thenReturn("ORCLPDB1");
+    // Act
+    ThrowingRunnable runnable = () -> AbstractOracleConnector.buildUrl(arguments);
+    // Assert
+    assertThrows(MetadataDumperUsageException.class, runnable);
+  }
+
+  @Test
+  public void buildUrl_providedNone_throwsException() {
+    ThrowingRunnable runnable = () -> AbstractOracleConnector.buildUrl(arguments);
+    assertThrows(MetadataDumperUsageException.class, runnable);
+  }
+
+  @Test
+  public void buildUrl_providedServiceName_success() {
+    when(arguments.getOracleServicename()).thenReturn("ORCLPDB");
+    when(arguments.getHost()).thenReturn("localhost");
+    when(arguments.getPort(anyInt())).thenReturn(1521);
+    // Act
+    String url = AbstractOracleConnector.buildUrl(arguments);
+    // Assert
+    assertEquals("jdbc:oracle:thin:@//localhost:1521/ORCLPDB", url);
+  }
+
+  @Test
+  public void buildUrl_providedSid_success() {
+    when(arguments.getOracleSID()).thenReturn("ORCLPDB1");
+    when(arguments.getHost()).thenReturn("localhost");
```

Comment:
I would suggest using a bit more unusual test data in at least one test (e.g. `sample-host-123`) to check that the `localhost` is passed from the arguments when constructing the URL and that it is not hardcoded. Similarly for port.

Reply:
IMO this is a slightly different piece of logic. The reason why I wanted to cover this is to verify that the URLs are built correctly for valid input data, without exceptions being thrown or malformed results, like missing a slash or colon, or a character added where it doesn't belong.

If we want to check that data is passed correctly from the cli args class to the connector classes, I don't mind adding it, but I'd write a separate test.

Reply:
Done, I've added a new test that provides uncommon host, port and sid values, and checks if they are get added to the built url.